Tests/AESEncrypterTests.swift

```diff
@@ -35,8 +35,73 @@ class AESEncrypterTests: RSACryptoTestCase {
         super.tearDown()
     }
 
+    /// Tests the `AES` encryption implementation for AES_128_CBC_HMAC_SHA_256 with the test data provided in the [RFC-7518](https://tools.ietf.org/html/rfc7518#appendix-B.1).
+    func testEncryptingA128CBCHS256() {
+        let plaintext = "41 20 63 69 70 68 65 72 20 73 79 73 74 65 6d 20 6d 75 73 74 20 6e 6f 74 20 62 65 20 72 65 71 75 69 72 65 64 20 74 6f 20 62 65 20 73 65 63 72 65 74 2c 20 61 6e 64 20 69 74 20 6d 75 73 74 20 62 65 20 61 62 6c 65 20 74 6f 20 66 61 6c 6c 20 69 6e 74 6f 20 74 68 65 20 68 61 6e 64 73 20 6f 66 20 74 68 65 20 65 6e 65 6d 79 20 77 69 74 68 6f 75 74 20 69 6e 63 6f 6e 76 65 6e 69 65 6e 63 65".hexadecimalToData()!
+        let additionalAuthenticatedData = "54 68 65 20 73 65 63 6f 6e 64 20 70 72 69 6e 63 69 70 6c 65 20 6f 66 20 41 75 67 75 73 74 65 20 4b 65 72 63 6b 68 6f 66 66 73".hexadecimalToData()!
+        let algorithm = SymmetricKeyAlgorithm.A128CBCHS256
+        let cek = try! SecureRandom.generate(count: algorithm.keyLength)
+        let encrypter = AESEncrypter(algorithm: algorithm)
+        let symmetricEncryptionContext = try! encrypter.encrypt(plaintext, with: cek, additionalAuthenticatedData: additionalAuthenticatedData.base64URLEncodedData())
+
+        // Check if the symmetric encryption was successful by using the CommonCrypto framework and not the implemented decrypt method.
+        let keys = try! algorithm.retrieveKeys(from: cek)
+        let hmacKey = keys.hmacKey
+        let encryptionKey = keys.encryptionKey
+
+        var concatData = String(data: additionalAuthenticatedData, encoding: .utf8)!.data(using: .utf8)!.base64URLEncodedData()
+        concatData.append(symmetricEncryptionContext.initializationVector)
+        concatData.append(symmetricEncryptionContext.ciphertext)
+        concatData.append(String(data: additionalAuthenticatedData, encoding: .utf8)!.data(using: .utf8)!.base64URLEncodedData().getByteLengthAsOctetHexData())
+
+        let keyLength = size_t(kCCKeySizeAES128)
+        var macOutData = Data(count: 64)
```

Comment:
``` suggestion
        var macOutData = Data(count: 32)
```

This is not too important as we only care about the first 16 bytes, but for correctness, this should be 32 (the output length in bytes of SHA256).